### PR TITLE
Better mapbox trace / layout-layer *below*

### PR DESCRIPTION
**Pull request by etpinard**
This PR resolves https://github.com/plotly/plotly.js/issues/4050 and resolves https://github.com/plotly/plotly.js/issues/4049

Commit https://github.com/plotly/plotly.js/commit/8a4e8f13657e3767b9b9aadde6a1aade23b10acf does most of the work here. Please carefully read the commit message for some rationale and tests. 

Commit https://github.com/plotly/plotly.js/commit/853a3919a65de29bf45e90bdc91801b613498d09 should help users debug. The other commits are cleanup commits.

cc @plotly/plotly_js @nicolaskruchten

### CODE REVIEW

File: src/traces/scattermapbox/plot.js [hunk 1]
```diff
@@ -50,23 +54,40 @@ proto.setSourceData = function(k, opts) {
...
 
 proto.update = function update(calcTrace) {
     var subplot = this.subplot;
+    var map = subplot.map;
     var optsAll = convert(calcTrace);
+    var below = subplot.belowLookup['trace-' + this.uid];
+    var i, k, opts;
+
+    if(below !== this.below) {
+        // console.log('update below', [below, this.below])
```
Comment (archmoj) [inline src/traces/scattermapbox/plot.js]: Thought you may want to remove this line.

Comment (etpinard) [reply src/traces/scattermapbox/plot.js]: Good eye. Thanks!

https://github.com/plotly/plotly.js/pull/4058/commits/39eff3a03cc76729d3b0163e09638e8b16b4f8be

File: src/traces/scattermapbox/attributes.js [hunk 2]
```diff
@@ -99,6 +99,18 @@ module.exports = overrideAll({
     textfont: mapboxAttrs.layers.symbol.textfont,
     textposition: mapboxAttrs.layers.symbol.textposition,
 
+    below: {
+        valType: 'string',
+        role: 'info',
+        description: [
+            'Determines if this scattermapbox trace\'s layers are inserted',
```
[the remark this answers is not in the document]
Comment (etpinard) [reply src/traces/scattermapbox/attributes.js]: New version in https://github.com/plotly/plotly.js/pull/4058/commits/39eff3a03cc76729d3b0163e09638e8b16b4f8be

**Comment by archmoj** [comment]
:dancer:

**Pull request merged**